### PR TITLE
feat(client):Weird validation for portfolio inputs in the setting page 

### LINKED ISSUE #49132: Weird validation for portfolio inputs in the setting page
### Describe the Issue

Validations are weirdly strict in imageUrl, but can still accept broken result, but lenient on URL section, although it doesn't accept broken URL.

### Affected Page

https://www.freecodecamp.org/settings

### Steps to Reproduce

1. Go to https://www.freecodecamp.org/settings
2. Type "https.com" in URL section
3. Type "https://github.com/freeCodeCamp/cdn/blob/main/build/chapter/puppy-small.jpg" in the image section
4. Go to your portfolio
5. See broken Media component

Note: I know that the correct URL is https://cdn.freecodecamp.org/chapter/orange-graphics-small.jpg, the issue is that I can spend time creating an item in portfolio, and it results in broken image later without knowing why.
image

### Expected behavior

- stricter validation on URL
- lenient validation on imageURL, maybe shows how image looks instead, so it becomes harder to break it in the portfolio page
image

### Screenshots

image

### System

- Device: [e.g. iPhone 6, Laptop]
- OS: [e.g. iOS 14, Windows 10, Ubuntu 20.04]
- Browser: [e.g. Chrome, Safari]
- Version: [e.g. 22]

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Anujvaid45**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Ref #49132 

- In this issue as mentioned, I have added a image preview in the add portfolio section 
- Due to this when a user enters a valid image URL, it will display a preview of the image 
- Also I have added CSS file for the same
- I have also locally tested this feature and it's results are in positive direction

### CODE REVIEW

**Comment by raisedadead** [comment]
Hi there, 

Thanks for creating this pull request.

Please ensure you follow the PR checklist and not just tick things off; otherwise, we won't take you seriously. Respect the requirements to earn the respect of your fellow maintainers. You will also save everyone time :)

Since you are new here, please pace yourself and read our [contributing guidelines](https://contribute.freecodecamp.org), as we see that you may still need to catch up on a few things.

<details>
<summary>Here are some examples (expand)</summary>
 
1. Do not edit files directly on GitHub – while you can, it's not a good idea. 

   Typos and formatting errors can break the tests.

2. Use the correct way to link issues. 

   Add the issue number only in the description of the PR by updating the `XXXXXX`. Do not just add issue numbers everywhere and anywhere you feel like.

4. Keep mentions and review requests to a minimum. 

   We understand you are excited about contributing, and our maintainers will get back to you when they can.

5. Do not work directly off your `main` branch. 

   You can always create a new branch for the changes you are working on. That way, you can sync change to your PR branch as the main repository moves ahead while your PR is waiting in the merge queue.

</details>

Don't worry. You don't have to close this PR. Feel free to ask specific queries on improving your PR here; someone will guide you.

We are happy you are excited to contribute and appreciate you taking the time to help us. Looking forward to more contributions!

Happy Contributing.

**Comment by Anujvaid45** [comment]
Hello @raisedadead 
- Sorry for missing few points in the contributing guidelines
- I have accordingly made changes
- Meanwhile my 2 tests are getting failed so currently I am working on it

File: package.json
Comment (naomi-lgbt) [inline package.json:102]: Why is stripe being installed in this PR?

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]

File: package.json
Comment (Anujvaid45) [reply package.json:102]: hello @naomi-lgbt 
I have not installed stripe in my repository and It already existed when I cloned the repository.

**Comment by Anujvaid45** [comment]
Hello @raisedadead 
Could you please help me out solving these failed tests?
I am not understanding what is to be done

File: package.json
Comment (ojeytonwilliams) [reply package.json:102]: Please could you revert the changes to the package.json and pnpm-lock.yaml. Neither file needs to have changed for this PR.

File: client/src/components/Header/components/auth-or-profile.tsx
Comment (naomi-lgbt) [inline client/src/components/Header/components/auth-or-profile.tsx:33]: Because you have updated the structure of the HTML here, you'll need to update the [related test](https://github.com/freeCodeCamp/freeCodeCamp/blob/2154b40ab623810ca609d6b7806a3fc20bf1528f/client/src/components/Header/header.test.tsx#L82) to look for the correct element in the DOM structure.

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
In addition to the tests, the package.json will need to be reverted.

File: client/src/components/settings/portfolio.css
Comment (Sboonny) [inline client/src/components/settings/portfolio.css:16]: The animation was causing my eyes to get watery when it was light mode, you can try to remake it, but I vote for deleting it, animation and polishing aspects can be done after this PR is merged

```suggestion
  border-radius: 5px;
}
```

File: client/src/components/settings/portfolio.css
Comment (Sboonny) [inline client/src/components/settings/portfolio.css:31]: This isn't providing a change to the image in the DOM
```suggestion
```

**Review by Sboonny — COMMENTED** [review]
I have change the comment from Close to Ref, because this do help a lot with the issue, but it doesn't close it

File: client/src/components/Header/components/auth-or-profile.tsx
Comment (Anujvaid45) [reply client/src/components/Header/components/auth-or-profile.tsx]: Hello @Sboonny 
- Done all the changes as said

File: client/src/components/settings/portfolio.css
Comment (Anujvaid45) [reply client/src/components/settings/portfolio.css]: Hello @Sboonny 
- Done the required changes

File: client/src/components/settings/portfolio.tsx
Comment (Anujvaid45) [reply client/src/components/settings/portfolio.tsx:290]: @Sboonny 
Here I am rendering it because if we don't do this then once we clear the image url, it will still display an empty image box which is not what we want

File: client/src/components/settings/portfolio.css
Comment (Anujvaid45) [reply client/src/components/settings/portfolio.css]: Done with required changes

**Comment by Anujvaid45** [comment]
Hello @Sboonny 
- As mentioned , I have done all the required changes
- However, the PR tests are continuously getting failed
- It would be nice if you could help me out with this

**Review by gikf — COMMENTED** [review]
Hey @Anujvaid45

I've fixed merge conflict for you.

There's one outstanding problem with the PR. Both `package.json` and `pnpm-lock.yaml` files need to be removed from the PR. There's couple ways to do that, the most straightforward at this time might be reverting the changes.

This can be also done with git. If you have `upstream` remote configured:
```
git checkout upstream/main -- package.json
git checkout upstream/main -- pnpm-lock.yaml
```
Then commit and push.

**Comment by Sboonny** [comment]
is it okay to push a commit that sort the conflicts @Anujvaid45?

**Comment by Anujvaid45** [comment]
> 



> is it okay to push a commit that sort the conflicts @Anujvaid45?

Sorry,Couldn't understand what are you trying to say

**Comment by Sboonny** [comment]
Do you want me to sort the issue with package.json and pnpm-lock?

**Comment by Anujvaid45** [comment]
> 

Please,It would be really helpful

File: client/src/components/settings/portfolio.tsx
Comment (Sboonny) [reply client/src/components/settings/portfolio.tsx:290]: Sorry for the delay about this, I am waiting for some feedback about how the image can be actually shown.

while we are waiting for the feedback, can you add alt message as following 

```ts
alt=`preview image for ${title}`
```

Comment (Anujvaid45) [reply client/src/components/settings/portfolio.tsx:290]: > Sorry for the delay about this, I am waiting for some feedback about how the image can be actually shown.
> 
> while we are waiting for the feedback, can you add alt message as following
> 
> ```ts
> alt=`A visual preview image for ${title}`
> ```

Okay ,will add that and push the change

Comment (Sboonny) [reply client/src/components/settings/portfolio.tsx:290]: no need, it isn't a necessary yet, because it's currently for visual users

Comment (Anujvaid45) [reply client/src/components/settings/portfolio.tsx:290]: What else is required to do?
Also all checks are passing now

**Comment by Anujvaid45** [comment]
hello @Sboonny 
Any update regarding the PR?

**Comment by Sboonny** [comment]
You can wrap the image in div and add the styles of the element to the wrapper to avoid the layout shifting, but still to avoid reworking and doing extra than needed, wait until a designer come up with mockup for this.

**Comment by Anujvaid45** [comment]
> You can wrap the image in div and add the styles of the element to the wrapper to avoid the layout shifting, but still to avoid reworking and doing extra than needed, wait until a designer come up with mockup for this.

@Sboonny 
Okay,Will be Waiting for the reply

**Comment by Sboonny** [comment]
Sorry @Anujvaid45, this should have been discussed on the issue to save your time, this is an oversight from me.

To solve the looks:

- Add preview image, it is a similar strategy to the profile image, so you can copy it without worry.
- Make the image or its surrounding wrapper to follow the same dimensions of the profile image card.

if you need help, feel free to mention so in the forum or discord.

**Comment by Sboonny** [comment]
Sorry @Anujvaid45, this should have been discussed on the issue to save your time, this is an oversight from me.

To solve the looks:

- Add preview image, it is a similar strategy to the profile image, so you can copy it without worry.
- Make the image or its surrounding wrapper to follow the same dimensions of the profile image card.

if you need help, feel free to mention so in the forum or discord.

**Comment by raisedadead** [comment]
Hi, 

We are closing old issues/pull requests that we think are stale. If you still need help or are fixing this, please open a new issue/pull request.

Thanks for your understanding. 

Happy contributing.

**Pull request closed without merging**
